### PR TITLE
DTRA-1192 / Kate/ DatePicker component

### PR DESCRIPTION
**Pull request by kate-deriv**
Created reusable atom component called DatePicker based on the React Calendar library.

        activeStartDate: 
                "The beginning of a period that shall be displayed. If you wish to use react-calendar in an uncontrolled way, use defaultActiveStartDate instead. Default value - today. Example values: new Date(2017, 0, 1).",

        allowPartialRange: 
                "Whether to call onChange with only partial result given selectRange prop. Default value - false.",

        calendarType: 
                "Type of calendar that should be used. Can be 'gregory', 'hebrew', 'islamic', 'iso8601'. Setting to 'gregory' or 'hebrew' will change the first day of the week to Sunday. Setting to 'islamic' will change the first day of the week to Saturday. Setting to 'islamic' or 'hebrew' will make weekends appear on Friday to Saturday. Default value - type of calendar most commonly used in a given locale.",

        className: 
                "Class name(s) that will be added along with 'react-calendar' to the main react-calendar <div> element.",

        defaultActiveStartDate:
        "The beginning of a period that shall be displayed by default. If you wish to use react-calendar in a controlled way, use activeStartDate instead. Default value - today. Example values: new Date(2017, 0, 1).",

        defaultValue: 
                "Calendar value that shall be selected initially. Can be either one value or an array of two values. If you wish to use react-calendar in a controlled way, use value instead. Example values: new Date(2017, 0, 1) or [new Date(2017, 0, 1), new Date(2017, 7, 1)].",

        defaultView: 
                "Determines which calendar view shall be opened initially. Does not disable navigation. Can be 'month', 'year', 'decade' or 'century'. If you wish to use react-calendar in a controlled way, use view instead. Default value - The most detailed view allowed.",

        isWidthFixed: 
                "Determines if width should be fixed or equal to container width. Default value - true",

        formatDay: 
                "Function called to override default formatting of day tile labels. Can be used to use your own formatting function. Default formatter is used if nothing was passed. Example: (locale, date) => formatDate(date, 'd')",

        formatLongDate: 
                "Function called to override default formatting of day tile abbr labels. Can be used to use your own formatting function. Default formatter is used if nothing was passed. Example: (locale, date) => formatDate(date, 'dd MMM YYYY')",

        formatMonth: 
                "Function called to override default formatting of month names. Can be used to use your own formatting function. Default formatter is used if nothing was passed. Example: (locale, date) => formatDate(date, 'MMM')",

        formatMonthYear: 
                "Function called to override default formatting of months and years. Can be used to use your own formatting function. Default formatter: (locale, date) => new Date(date).toLocaleString(locale || navigator.languages, { month: 'short', year: 'numeric' })`. Example: (locale, date) => formatDate(date, 'MMMM YYYY')",

        formatShortWeekday:
                "Function called to override default formatting of weekday names (shortened). Can be used to use your own formatting function. Default formatter is used if nothing was passed. Example: (locale, date) => formatDate(date, 'dd')",

        formatWeekday: 
                "Function called to override default formatting of weekday names. Can be used to use your own formatting function. Default formatter is used if nothing was passed. Example: (locale, date) => formatDate(date, 'dd')",

        formatYear: 
                "Function called to override default formatting of year in the top navigation section. Can be used to use your own formatting function. Default formatter is used if nothing was passed. Example: (locale, date) => formatDate(date, 'YYYY')",

        goToRangeStartOnSelect: 
                "Whether to go to the beginning of the range when selecting the end of the range. Default value - true.",
 
        inputRef: 
                "A prop that behaves like ref, but it's passed to main 'div' rendered by 'Calendar' component.",

        locale: 
                "Locale that should be used by the calendar. Can be any IETF language tag. Note: When using SSR, setting this prop may help resolving hydration errors caused by locale mismatch between server and client. Default value: server locale/User's browser settings. Example: 'hu-HU'.",

        maxDate: 
                "Maximum date that the user can select. Periods partially overlapped by maxDate will also be selectable, although react-calendar will ensure that no later date is selected. Example values: new Date().",

        maxDetail: 
                "The most detailed view that the user shall see. View defined here also becomes the one on which clicking an item will select a date and pass it to onChange. Can be 'month', 'year', 'decade' or 'century'. Default value - 'month'.",

        minDate: 
        "Minimum date that the user can select. Periods partially overlapped by minDate will also be selectable, although react-calendar will ensure that no earlier date is selected. Example values: new Date().",

        minDetail: 
                "The least detailed view that the user shall see. Default value - 'century'.",

        navigationAriaLabel: 
                "aria-label attribute of a label rendered on calendar navigation bar. Example: 'Go up'.",

        navigationAriaLive: 
                "aria-live attribute of a label rendered on calendar navigation bar. Example: 'polite'.",

        navigationLabel: 
                "Content of a label rendered on calendar navigation bar.",

        next2AriaLabel: 
                "aria-label attribute of the 'next on higher level' button on the navigation pane. Example: 'Jump forwards'",

        next2Label: 
                "Content of the 'next on higher level' button on the navigation pane. Setting the value explicitly to null will hide the icon. Default value: null. Example: String: '»'",

        nextAriaLabel: 
                "aria-label attribute of the 'next' button on the navigation pane. Example: 'Next'",

        nextLabel: 
                "Content of the 'next' button on the navigation pane. Setting the value explicitly to null will hide the icon. Default value: LabelPairedChevronRightSmFillIcon. Example: String: '>'",

        optionsConfig: 
                "Config for changing format of the selected date, will be passed inside of Date.toLocaleDateString(). Default value: {day: '2-digit', month: '2-digit', year: 'numeric'}",

        onActiveStartDateChange: 
                "Function called when the user navigates from one view to another using previous/next button. Note that this function will not be called when e.g. drilling up from January 2021 to 2021 or drilling down the other way around. action signifies the reason for active start date change and can be one of the following values: 'prev', 'prev2', 'next', 'next2', 'drillUp', 'drillDown', 'onChange'. Example: ({ action, activeStartDate, value, view }) => alert('Changed view to: ', activeStartDate, view)",

        onChange: 
                "Function called when the user clicks an item (day on month view, month on year view and so on) on the most detailed view available.",

        onFormattedDate: 
                "Function called when the user clicks an item (day on month view, month on year view and so on) on the most detailed view available, will be called together with onChange. Returns formatted chosen date (single or range).",

        prev2AriaLabel: 
                "aria-label attribute of the 'previous on higher level' button on the navigation pane. Example: 'Jump backwards'",

        prev2Label: 
                "Content of the 'previous on higher level' button on the navigation pane. Setting the value explicitly to null will hide the icon. Default value: null. Example: String: '«'",

        prevAriaLabel: 
                "aria-label attribute of the 'previous' button on the navigation pane. Example: 'Previous'",

        prevLabel: 
                "Content of the 'previous' button on the navigation pane. Setting the value explicitly to null will hide the icon. Default value: LabelPairedChevronLeftSmFillIcon. Example: String: '<'",

        returnValue: 
                "Which dates shall be passed by the calendar to the onChange function and onClick{Period} functions. Can be 'start', 'end' or 'range'. The latter will cause an array with start and end values to be passed. Default value: 'start'.",

        selectRange:
                "Whether the user shall select two dates forming a range instead of just one. Note: This feature will make react-calendar return array with two dates regardless of returnValue setting. Default value: false.",

        showNavigation: 
                "Whether a navigation bar with arrows and title shall be rendered. Default value: true.",

        showNeighboringMonth: 
                "Whether days from previous or next month shall be rendered if the month doesn't start on the first day of the week or doesn't end on the last day of the week, respectively. Default value: false.",

        tileClassName:
                "Class name(s) that will be applied to a given calendar item (day on month view, month on year view and so on).",

        value: 
                "Calendar value. Can be either one value or an array of two values. If you wish to use react-calendar in an uncontrolled way, use defaultValue instead.",

        wrapperClassName: 
        "Class that will be applied to a calendar wrapper",

Screenshot 2024-05-10 at 11 25 27 AM
Screenshot 2024-05-10 at 11 25 18 AM

### CODE REVIEW

File: lib/components/Atom/Calendar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    isWidthFixed?: boolean;
+    wrapperClassName?: string;
+}
+
+type ValuePiece = Date | null;
+
+type Value = ValuePiece | [ValuePiece, ValuePiece];
+
+export const DatePicker = ({
+    activeStartDate,
```
Comment (prince-deriv) [inline lib/components/Atom/Calendar/index.tsx:29]: Any difference if we keep this `startDate` ?

Comment (kate-deriv) [reply lib/components/Atom/Calendar/index.tsx:29]: No, it's okay to rename, I'll push the commit

File: lib/components/Atom/Calendar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+type Value = ValuePiece | [ValuePiece, ValuePiece];
+
+export const DatePicker = ({
+    startDate,
+    allowPartialRange = false,
+    calendarType,
+    className,
+    defaultValue,
+    defaultView,
+    isWidthFixed = true,
```
Comment (maryia-deriv) [inline lib/components/Atom/Calendar/index.tsx:39]: it's up to you but what do you think about naming this prop `hasFixedWidth` ? :)

File: lib/components/Atom/Calendar/date-picker.scss [hunk 3]
```diff
@@ -0,0 +1,177 @@
+@import "@quill/quill.scss";
```
Comment (kate-deriv) [inline lib/components/Atom/Calendar/date-picker.scss:1]: Without this import, styles were not loaded on the first render, only after switching to another story/components. Not sure what I could do else, seems something wrong with settings/configuration

Comment (kate-deriv) [reply lib/components/Atom/Calendar/date-picker.scss:1]: cc: @prince-deriv

File: lib/components/Atom/Calendar/index.tsx (hunk 2 above)
Comment (kate-deriv) [reply lib/components/Atom/Calendar/index.tsx:39]: Sure, sounds better that current prop) Will push the fix

File: lib/components/Atom/Calendar/date-picker.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,583 @@
...
+export const DatePickerWithNeighboringMonth = Template.bind(this) as Story;
+DatePickerWithNeighboringMonth.args = {
+    ...meta.args,
+    showNeighboringMonth: true,
+};
+
+export const DatePickerWithCustomActiveStartDate = Template.bind(this) as Story;
+DatePickerWithCustomActiveStartDate.args = {
+    ...meta.args,
+    startDate: new Date(2017, 0, 1),
```
Comment (maryia-deriv) [inline lib/components/Atom/Calendar/date-picker.stories.tsx:564]: what do you think about `new Date('01/01/2017')` ? :)

Comment (kate-deriv) [reply lib/components/Atom/Calendar/date-picker.stories.tsx:564]: Sure, why not, let me push the commit

**Review by maryia-deriv — APPROVED** [review]

**Review by balakrishna-deriv — APPROVED** [review]

**Review by prince-deriv — APPROVED** [review]

**Pull request merged**